### PR TITLE
ENG-497: rebuild the cms publication widget screen

### PR DESCRIPTION
**Pull request by alepintus**

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **nicpuddu**

If they are not already assigned, you can assign the PR to them by writing `/assign @nicpuddu` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fentando-cms).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/entando-cms/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by entando-jx** [comment]
@alepintus: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 364f595238aeb1c0ad54c31fc40b4f0cec50eb57 | [link](https://raw.githubusercontent.com/entando/devops-results/logs/jenkins-x/logs/entando/entando-cms/PR-177/1.log) | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by alepintus** [comment]
Looks good

File: src/sass/widgets/forms/ContentsFilterModal.scss [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#ContentsFilterModal { // sass-lint:disable-line no-ids
+  .modal-dialog { // sass-lint:disable-line class-name-format
+    width: 80%;
+  }
+}
```
Comment (gidesan) [inline src/sass/widgets/forms/ContentsFilterModal.scss:5]: please add a newline at the end of the file

File: src/sass/widgets/forms/ContentsFilterModal.scss [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#ContentsFilterModal { // sass-lint:disable-line no-ids
```
Comment (gidesan) [inline src/sass/widgets/forms/ContentsFilterModal.scss:1]: do we really need to use an ID?

File: src/sass/widgets/forms/SingleContentConfigFormBody.scss [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  }
+
+  &__templateTitle { // sass-lint:disable-line class-name-format
+    margin-top: 15px;
+  }
+}
+
+form.SingleContentConfigForm { // sass-lint:disable-line no-qualifying-elements force-element-nesting
+  background-color: white;
+}
```
Comment (gidesan) [inline src/sass/widgets/forms/SingleContentConfigFormBody.scss:20]: please add a newline at the end of the file

File: src/state/contents/actions.js [hunk 4]
```diff
@@ -18,12 +18,16 @@ import {
...
   SET_CURRENT_STATUS_SHOW, SET_CURRENT_COLUMNS_SHOW, SET_SORT, SET_CONTENT_TYPE, SET_TAB_SEARCH,
   SET_GROUP, SELECT_ROW, SELECT_ALL_ROWS, SET_JOIN_CONTENT_CATEGORY, RESET_JOIN_CONTENT_CATEGORIES,
-  RESET_AUTHOR_STATUS,
+  RESET_AUTHOR_STATUS, SELECT_SINGLE_ROW, CLEAR_CONTENTS_STATE,
 } from 'state/contents/types';
 import { postAddContent } from 'api/editContent';
 
 const pageDefault = { page: 1, pageSize: 10 };
 
+export const clearContentsState = () => ({
```
Comment (gidesan) [inline src/state/contents/actions.js:27]: can this have a name that is closer to what the user does?

File: src/state/contents/reducer.js [hunk 5]
```diff
@@ -123,6 +125,16 @@ const reducer = (state = defaultState, action = {}) => {
         sortingColumns: action.payload,
       };
     }
+    case SELECT_SINGLE_ROW: {
+      const row = action.payload;
+      const selectedRows = [];
+      selectedRows.push(row.id);
```
Comment (gidesan) [inline src/state/contents/reducer.js:131]:
```suggestion
      const selectedRows = [ lastSelectedRow.id ];
```

File: src/state/contents/reducer.js [hunk 6]
```diff
@@ -123,6 +125,16 @@ const reducer = (state = defaultState, action = {}) => {
         sortingColumns: action.payload,
       };
     }
+    case SELECT_SINGLE_ROW: {
+      const row = action.payload;
```
Comment (gidesan) [inline src/state/contents/reducer.js:129]:
```suggestion
      const lastSelectedRow = action.payload;
```

File: src/state/contents/reducer.js [hunk 7]
```diff
@@ -123,6 +125,16 @@ const reducer = (state = defaultState, action = {}) => {
...
+      const selectedRows = [];
+      selectedRows.push(row.id);
+      return {
+        ...state,
+        selectedRows,
+        lastSelectedRow: row,
```
Comment (gidesan) [inline src/state/contents/reducer.js:135]:
```suggestion
        lastSelectedRow,
```

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando/entando-cms/pull/177#pullrequestreview-423682038" title="Approved">gidesan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fentando-cms).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/entando/entando-cms/blob/master/OWNERS)~~ [gidesan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**